src/adaptive_graph_of_thoughts/tests/test_graph_elements.py

```
@@ -0,0 +1,103 @@
+    "invalid_input",
+    [None, 123, 3.14, [], {}],
+)
+def test_invalid_node_input_raises_type_error(empty_graph, invalid_input):
+    """Failure condition: add_node with invalid input raises TypeError."""
+    g = empty_graph
+    with pytest.raises(TypeError):
+        g.add_node(invalid_input)
+
+def test_node_equality_and_attributes():
+    """Unit test: Node class initialization and equality."""
+    n1 = Node('id1')
```

Ruff (0.11.9)

Undefined name `Node`

(F821)